### DISCUSSION TITLE
looking for instructions on how to generate .lydrc and .lylvs

### DISCUSSION THREAD

**Discussion by gkamendje** [The-OpenROAD-Project/OpenROAD-flow-scripts#1450] [General] [answered]
I was wondering if someone could point me to any usefull resource on how to generate `.lydrc` and `.lylvs` scripts used by KLayout for DRC and LVS. In particular, I was wondering if there is any script around that could be used to generate them outgoing from `.LEF` files or similar "standard" technology files. I could not find anything on the KLayout site/forum. I had a look at the `KLayout DRC -Flat File documentation` but that looks rather scary to me at this point.

**Comment by rovinski** [comment, marked as answer]
DRC guide: https://www.klayout.de/doc/manual/drc_runsets.html
DRC reference and API: https://www.klayout.de/doc-qt5/about/drc_ref.html
LVS overview: https://www.klayout.de/doc-qt5/manual/lvs_overview.html
LVS reference and API: https://www.klayout.de/doc-qt5/about/lvs_ref.html
As noted, most of the LVS API is part of the DRC API and only LVS-specific features are documented there.

Laurent is the original author of many of the widely-used public KLayout DRC/LVS scripts and you can find them on their profile: https://github.com/laurentc2
There are good examples of practical rule checks there.

**Comment by rovinski** [comment]
There is no automated tool that I know of which can convert tech LEF rules to KLayout checks. If someone made one, it would be a fantastic contribution to the community.

Do note that usually design rules are created by the foundry first based on actual process constraints and then translated to LEF from there. Because LEF lacks the verbosity that some checks require, the actual rule is simplified in LEF to (usually) a more conservative check. If you were to translate from LEF back into raw DRC rules, it should be valid but not necessarily as liberal as the original rules. That being said, automatically generating an initial template for the backend rules would still be hugely helpful.

**Reply by gkamendje** [reply]
Thanks for the useful info. I will contact Laurent and see if there is anything to be done.
